include/mlio/image_reader.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You
+ * may not use this file except in compliance with the License. A copy of
+ * the License is located at
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <opencv4/opencv2/core/mat.hpp>
+
+#include "mlio/config.h"
+#include "mlio/parallel_data_reader.h"
+
+namespace mlio {
+inline namespace v1 {
+
+/// @addtogroup data_readers Data Readers
+/// @{
+
+enum class image_frame { file, pipe };
+
+struct MLIO_API image_reader_params final {
+    /// Enum to identify 'file' or 'pipe' mode.
+    image_frame img_frame{};
```

Comment:
Are we breaking any abstraction barriers by requiring the image reader to know the input mode of the data? How does this compare to other data readers?

Reply:
hmm I don't think so. Do you suggest not having the image_frame param? Also we will default to `file` for python API.

As for the comparison, this is similar to how we have csv params: https://github.com/awslabs/ml-io/blob/master/include/mlio/csv_reader.h#L49

Reply:
It doesn't seem like this is necessary for any other data reader though. Why is it that in this case, a difference in data source changes how the data reader needs to behave? Specifically, [this](https://github.com/rizwangilani/ml-io/blob/imreader/src/mlio/image_reader.cxx#L108-L113) is the only place where `img_frame` is used, and I'm wondering if there's a way to handle this before it gets to the image reader.

Reply:
we are also using `img_frame` to identify what type of `record_reader` to we want to build in `make_record_reader()`. Alternatively, we can create 2 different image readers. however, that might be an overkill at this point. thoughts?